pkg/kp/pcstore/store.go

```diff
@@ -68,8 +69,35 @@ type ConcreteSyncerType string
 func (t ConcreteSyncerType) String() string { return string(t) }
 
 type ConcreteSyncer interface {
+	// SyncCluster implements a concrete synchronization of the pod cluster to
+	// some real world implementation of a load balancer, credential management
+	// system, or service discovery implementation.
+	//
+	// When a ConcreteSyncer is passed to WatchAndSync, SyncCluster is called
+	// every time the set of labeled pods change, the pod cluster's metadata
+	// changes, or when the long-lived watch on the pod cluster store returns.
+	// This function is expected to be idempotent. Returned errors do not change
+	// the rules for invocation of this function.
+	//
+	// SyncCluster will be called for every pod cluster present in the store.
+	//
+	// ConcreteSyncers will be called concurrently and must operate safely.
 	SyncCluster(pc *fields.PodCluster, pods []labels.Labeled) error
+
+	// DeleteCluster is called when a pod cluster is observed to have been removed
+	// from the store. DeleteCluster can be invoked in two circumstances: first,
+	// if the cluster was present in a first watch, then absent in a subsequent watch,
+	// then it is assumed it was deleted from the pcstore. Second, if the call
+	// to GetInitialClusters() returns a pod cluster ID that is not present in the very
+	// first watch result, DeleteCluster will be invoked with that pod cluster. In the
+	// latter case, only the pod cluster's ID will be set on the passed `pc` object.
```

Comment:
`DeleteCluster` should probably ONLY ever receive the ID. depending on anything other than the ID would be a bug